facade/src/main/java/com/redhat/lightblue/migrator/facade/ServiceFacade.java

```diff
@@ -94,6 +105,21 @@ public ServiceFacade(D legacySvc, D lightblueSvc, String implementationName, Pro
 
         timeoutConfiguration = new TimeoutConfiguration(DEFAULT_TIMEOUT_MS, implementationName, this.properties);
 
+        // create ThreadPoolExecutor for all facades
+        synchronized("executor") {
```

Comment:
> Unless the string "executor" is a single instance shared this is not going to correctly protect this block of code. Pick instead some static constant on this class, such as CONFIG_PREFIX, to synchronize on.

It is a single instance, no? The simple syntax (no new keyword) will always use the same object. Example:
```
System.out.println("foo" == "foo");
System.out.println(new String("foo") == new String("foo"));

true
false
```

I'll add that additional null check.